awx/sso/pipeline.py

```diff
@@ -255,17 +255,18 @@ def _check_flag(user, flag, attributes, user_flags_settings):
             # If so, check and see if the value of the attr matches the required value
             attribute_value = attributes.get(attr_setting, None)
             if isinstance(attribute_value, (list, tuple)):
-                attribute_value = attribute_value[0]
-            if attribute_value == user_flags_settings.get(is_value_key):
-                logger.debug("Giving %s %s from attribute %s with matching value" % (user.username, flag, attr_setting))
-                new_flag = True
-            # if they don't match make sure that new_flag is false
-            else:
-                logger.debug(
-                    "Refusing %s for %s because attr %s (%s) did not match value '%s'"
-                    % (flag, user.username, attr_setting, attribute_value, user_flags_settings.get(is_value_key))
-                )
-                new_flag = False
+                for attribute_value in attribute_value:            
+                    if attribute_value == user_flags_settings.get(is_value_key):
+                        logger.debug("Giving %s %s from attribute %s with matching value" % (user.username, flag, attr_setting))
+                        new_flag = True
+                        break
+                    # if they don't match make sure that new_flag is false
+                    else:
+                        logger.debug(
+                            "Refusing %s for %s because attr %s (%s) did not match value '%s'"
+                            % (flag, user.username, attr_setting, attribute_value, user_flags_settings.get(is_value_key))
+                        )
+                        new_flag = False
```

Comment:
This else block was previously run outside of a loop. I don't think we would want to display that debug message for each of the values in an array (especially if one of them passes). Perhaps we could move this out of the for loop and do something like:
```
new_flag = False
<for loop>
    <has attribute if block>
    
if not new_flag:
    logger.debug(
        "Refusing %s for %s because attr %s (%s) did not match value '%s'"
        % (flag, user.username, attr_string, ', '.join(attribute_values, user_flags_settings.get(is_value_key))
    )
```